tasks/main.yml

```diff
@@ -0,0 +1,169 @@
+---
+- name: Import dovecot key
+  apt_key:
+    url: https://repo.dovecot.org/DOVECOT-REPO-GPG
+
+- name: Add dovecot repository
+  apt_repository:
+    repo: deb https://repo.dovecot.org/ce-{{ dovecot_release }}-latest/debian/{{ ansible_distribution_release }} {{ ansible_distribution_release }} main
+    filename: dovecot
+
+- name: Install dovecot dependencies
+  apt:
+    name:
+      - dovecot-core
+      - dovecot-imapd
+      - dovecot-ldap
+      - dovecot-lucene
+      - dovecot-managesieved
+      - dovecot-sieve
+
+- name: Create configuration file
+  template:
+    src: dovecot.conf.j2
+    dest: /etc/dovecot/dovecot.conf
+    owner: root
+    group: root
+    mode: 0640
+  notify: Reload dovecot
+
+- name: Create LDAP passdb configuration
+  template:
+    src: ldap.conf.j2
+    dest: /etc/dovecot/ldap-passdb.conf
+    owner: root
+    group: root
+    mode: 0640
+  notify: Reload dovecot
+
+# I could use the same file, but the documentation
+# suggests different filenames to ensure different
+# connections are used which boosts performance as
+# the userdb lookups can be made async.
+- name: Create LDAP userdb configuration
+  file:
+    src: ldap-passdb.conf
+    dest: /etc/dovecot/ldap-userdb.conf
+    state: link
+    owner: root
+    group: root
+  when: dovecot_dsync
+  notify: Reload dovecot
+
+- name: Create LDAP master configuration
+  template:
+    src: ldap-master.conf.j2
+    dest: /etc/dovecot/ldap-master.conf
+    owner: root
+    group: root
+    mode: 0640
+  notify: Reload dovecot
+
+- name: Create virtual mail group
+  group:
+    name: "{{ dovecot_vmail_group }}"
+    system: true
+
+- name: Create virtual mail user
+  user:
+    name: "{{ dovecot_vmail_user }}"
+    group: "{{ dovecot_vmail_group }}"
+    comment: Virtual mail user
+    home: /var/empty
+    create_home: false
+    shell: /bin/false
+    system: true
+
+- name: Ensure permissions of replicator database
+  file:
+    dest: /var/lib/dovecot/replicator.db
+    state: touch
+    mode: 0600
+    owner: "{{ dovecot_vmail_user }}"
+    group: "{{ dovecot_vmail_group }}"
+  changed_when: false
+  when: dovecot_dsync
+  notify: Reload dovecot
```

Comment:
I don't think this works if you set changed_when to false

Reply:
True that